### PR TITLE
feat: Remove runners and ingress from dal GetStatus

### PR DESCRIPTION
**Pull request by deniseli**
Fixes issue https://github.com/TBD54566975/ftl/issues/1171

This PR includes some proto field deletions following the guidance here: https://protobuf.dev/programming-guides/proto3/#deleting

Manually tested `ftl serve` and `ftl status` (the two commands that reference `StatusRequest`) and got the desired output:
```
$ ftl status
{
  "controllers": [
    {
      "key": "ctr-0-3lky7r59cey54xq",
      "endpoint": "http://localhost:8892"
    }
  ]
}% 
$ ftl serve --recreate
info: Starting FTL with 1 controller(s)
info:controller0: Web console available at: http://localhost:8892
^C% 
$
```

### CODE REVIEW

**Review by worstell — APPROVED** [review]

**Review by alecthomas — COMMENTED** [review]
Sorry, I guess the ticket wasn't clear - what it meant was to get rid of the ability to select _all_ controllers/runners, etc. not get rid of the ability to retrieve active runners/controllers altogether. Please revert and adjust.

File: backend/protos/xyz/block/ftl/v1/ftl.proto [hunk 1]
```diff
@@ -184,9 +184,9 @@ message StreamDeploymentLogsRequest {
 message StreamDeploymentLogsResponse {}
 
 message StatusRequest {
-  bool all_runners = 1;
   bool all_controllers = 2;
-  bool all_ingress_routes = 3;
+
+  reserved 1, 3;
```
Comment (alecthomas) [inline backend/protos/xyz/block/ftl/v1/ftl.proto:189]: Remove this, as Wes mentioned, we don't bother with preserving proto compatibility.

File: backend/protos/xyz/block/ftl/v1/ftl.proto [hunk 2]
```diff
@@ -217,23 +207,15 @@ message StatusResponse {
...
-
   message Route {
     string module = 1;
     string runner = 2;
     string deployment = 3;
     string endpoint = 4;
   }
   repeated Route routes = 5;
+
+  reserved 2, 4;
```
Comment (alecthomas) [inline backend/protos/xyz/block/ftl/v1/ftl.proto:218]: Same here.

File: backend/protos/xyz/block/ftl/v1/ftl.proto (hunk 1 above)
Comment (deniseli) [reply backend/protos/xyz/block/ftl/v1/ftl.proto:189]: got it, makes sense

File: backend/protos/xyz/block/ftl/v1/ftl.proto (hunk 2 above)
Comment (deniseli) [reply backend/protos/xyz/block/ftl/v1/ftl.proto:218]: got it, makes sense

**Pull request merged**